src/stan/error_handling/matrix/check_nonzero_size.hpp

```diff
@@ -14,33 +14,36 @@ namespace stan {
 
     /**
      * Return <code>true</code> if the specified matrix/vector is of
-     * non-zero size. Throws a std:domain_error otherwise. The message
+     * non-zero size. 
+     *
+     * Throws a std:invalid_argument otherwise. The message
      * will indicate that the variable name "has size 0".
      *
      * NOTE: this will not throw if y contains nan values.
      *
-     * @param function
-     * @param y matrix/vector to test against
-     * @param name
+     * @tparam T_y Type of container
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Container to test. This will accept matrices and vectors
+     *
      * @return <code>true</code> if the the specified matrix/vector is 
      * of non-zero size
-     * @throw std::domain_error if the specified matrix/vector is of
-     * non-zero size
-     * @tparam T Type of scalar.
+     * @throw <code>std::invalid_argument</code> if the specified matrix/vector 
+     *   has zero size
      */
     template <typename T_y>
     inline bool check_nonzero_size(const std::string& function,
                                    const std::string& name,
                                    const T_y& y) {
-      using stan::math::index_type;
-
-      typedef typename index_type<T_y>::type size_t;
       if (y.size() > 0) 
         return true;
 
-      dom_err(function, name, size_t(), 
-              "has size ", 
-              ", but must have a non-zero size");
+      using stan::math::index_type;
+      typedef typename index_type<T_y>::type size_t;
+      invalid_argument(function, name, size_t(), 
```

Comment:
I don't see what `size_t()` is doing.  Can't we put 0 in there explicitly if you want 0 to come out?